modules/tinymce/src/core/main/ts/focus/FocusController.ts

```diff
@@ -95,14 +96,16 @@ const registerEvents = function (editorManager: EditorManager, e: { editor: Edit
     documentFocusInHandler = function (e: FocusEvent) {
       const activeEditor = editorManager.activeEditor;
 
-      const target = e.target as HTMLElement;
-
-      if (activeEditor && target.ownerDocument === document) {
-        // Fire a blur event if the element isn't a UI element
-        if (target !== document.body && !isUIElement(activeEditor, target) && editorManager.focusedEditor === activeEditor) {
-          activeEditor.fire('blur', { focusedEditor: null });
-          editorManager.focusedEditor = null;
-        }
+      if (activeEditor) {
+        ShadowDom.getOriginalEventTarget(e).each((target: Element) => {
+          if (target.ownerDocument === document) {
```

Comment:
How will this work? As I would imagine the `ownerDocument` would be the shadow root, not `document` so in most cases this would be false.

Reply:
`ownerDocument` of an element in a shadow root is the document. `element.getRootNode()` returns the shadow root or document.

Reply:
Interesting and rather unexpected... thanks for clarifying!